exercises/space-age/.meta/hints.md

```diff
@@ -4,3 +4,10 @@ Some Rust topics you may want to read about while solving this problem:
 
 - Traits, both the From trait and implementing your own traits
 - Default method implementations for traits
+- Macros, the use of a macro could reduce boilerplate and increase readability 
+  for this exercise. For instance, 
+  [a macro can implement a trait for multiple types at once](https://stackoverflow.com/questions/39150216/implementing-a-trait-for-multiple-types-at-once),
+  though it is fine to implement `years_during` in the Planet trait itself. A macro could
+  define both the structs and their implementations. Info to get started with macros can 
+  be found [here](https://doc.rust-lang.org/rust-by-example/macros/syntax.html) and 
+  [here](https://doc.rust-lang.org/1.7.0/book/macros.html).
```

Comment:
Why did you pick the version 1.7.0 edition of the book? That's long obsolete; stable is version 1.49 these days.

Reply:
In comparing the two, I think I prefer the step-by-step explanation in the 1.7.0 version that starts with defining a macro compared to the most recent version that starts with the difference between macros and functions. The 1.7.0 also has a `syntactic requirements` section not found on the new version page. And, the older version covers mostly declarative macros, which are the kind intended for space-age; the new version covers more about procedural macros.

Reply:
If we're going to use an old version, then let's ensure that we maintain these properties:

- prioritize the link to the most current documentation: https://doc.rust-lang.org/stable/book/ch19-06-macros.html
- link to the most recent edition of the first edition chapter: <https://doc.rust-lang.org/1.30.0/book/first-edition/macros.html>
- point out in the hint what precisely is useful in the older edition of the book

One more point related to this line, but for the other link: the Rust by Example link should probably go to the declarative macros overview (<https://doc.rust-lang.org/stable/rust-by-example/macros.html>) instead of the much sparser syntax page.

Reply:
Okay, thanks, I've incorporated your changes and committed them locally and pushed them to my my fork. Don't know how to update the earlier PR here with the changes from the latest commit to my fork. VSCode made both commits on my master. I'm gonna research in a book on Git I have, but if I don't figure it out I may need your guidance.

```
 be found at:
  [The Macros chapter in The Rust Programming Language](https://doc.rust-lang.org/stable/book/ch19-06-macros.html)
  [an older version of the Macros chapter with helpful detail](https://doc.rust-lang.org/1.30.0/book/first-edition/macros.html).
  [Rust By Example](https://doc.rust-lang.org/stable/rust-by-example/macros.html)
  ```

Reply:
Okay, it looks like this PR updated with my most recent b038... commit as seen below.